modules/shared/src/main/scala/org/tessellation/currency/schema/KeyMessage.scala

```diff
@@ -0,0 +1,46 @@
+package org.tessellation.currency.schema
+
+import java.security.KeyPair
+
+import cats.effect.Async
+import cats.syntax.either._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import org.tessellation.schema.address.{Address, DAGAddressRefined}
+import org.tessellation.security.signature.Signed
+import org.tessellation.security.{Hasher, SecurityProvider}
+
+import derevo.cats.{eqv, show}
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
+import eu.timepit.refined.auto._
+import eu.timepit.refined.refineV
+import io.estatico.newtype.macros.newtype
+
+object KeyMessage {
+  @derive(eqv, show, encoder, decoder)
+  @newtype
+  case class AdminFeeKeyMessage(value: Signed[Address])
+
+  object AdminFeeKeyMessage {
+    def apply[F[_]: Async: SecurityProvider: Hasher](s: String, keyPair: KeyPair): F[AdminFeeKeyMessage] =
+      toSignedAddress(s, keyPair).map(AdminFeeKeyMessage(_))
+  }
+
+  @derive(eqv, show, encoder, decoder)
+  @newtype
+  case class StakingMessage(value: Signed[Address])
+
+  object StakingMessage {
+    def apply[F[_]: Async: SecurityProvider: Hasher](s: String, keyPair: KeyPair): F[StakingMessage] =
+      toSignedAddress(s, keyPair).map(StakingMessage(_))
+  }
+
+  private def toSignedAddress[F[_]: Async: SecurityProvider: Hasher](s: String, keyPair: KeyPair): F[Signed[Address]] =
```

Comment:
This seems correct to me, and this is the output that we want to print out when a user runs the appropriate CLI command.